### PR TITLE
Improve webpack "exclude" expressions

### LINKED ISSUE #21662: Webassets can not be built when the repo is under a certain path
Expected behavior: Frontend assets build successfully, no matter what path the repository is checked out under.

Current behavior: If the path that the Teleport repository is checked out under a directory that has any folder whose name includes the string `assets`, Webpack will mistakenly exclude `tsx` source files under consideration, and fail to build the frontend assets.

This is due to a lax regular expression that defines paths to exclude:

https://github.com/gravitational/teleport/blob/b1a29ce99a00e1a4d41989d0abdbad5f97fb31c1/web/packages/build/webpack/webpack.base.js#L104

Although parsing paths with regular expressions is somewhat suboptimal, I understand that it is conventional in the Webpack ecosystem. A stricter regexp, that would match an exact name of a single ancestor, as used here:

https://github.com/gravitational/teleport/blob/b1a29ce99a00e1a4d41989d0abdbad5f97fb31c1/web/packages/build/webpack/webpack.base.js#L134

seems to solve the problem.

Bug details:
- Teleport version: v11, master
- Recreation steps:
```console
$ git clone --depth 1 https://github.com/gravitational/teleport assets-bug-repro # check out under a dir that includes "assets"
$ cd assets-bug-repro
$ yarn install
$ yarn build-ui-oss
yarn run v1.22.19
$ yarn workspace @gravitational/teleport build --output-path=../../../webassets/teleport/app
$ g-build --config webpack.prod.config.js --progress --bail --output-path=../../../webassets/teleport/app
assets by status 3.92 KiB [cached] 4 assets
Entrypoint app = runtime~app.d96ed60b58b56df22960.js app.33babda0435fe6278445.js
runtime modules 2.5 KiB 3 modules
./src/boot.tsx 1.22 KiB [built] [code generated] [1 error]

ERROR in ./src/boot.tsx 41:2
Module parse failed: Unexpected token (41:2)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|
| ReactDOM.render(
>   <Teleport history={history.original()} ctx={teleportContext} />,
|   document.getElementById('app')
| );

webpack 5.73.0 compiled with 1 error in 360 ms
Done in 1.76s.
```

### PR DESCRIPTION
**Pull request by justinas**
Fixes https://github.com/gravitational/teleport/issues/21662 by matching on the exact folder name `node_modules` (as we do in [another place](https://github.com/gravitational/teleport/issues/21662)). The existing version excludes anything that has substring `assets` or `node_modules` in any part of the full path.

### CODE REVIEW

File: web/packages/build/webpack/webpack.base.js [hunk 1]
```diff
@@ -104,7 +104,7 @@ const configFactory = {
     jsx() {
       return {
         test: /\.(ts|tsx|js|jsx)$/,
-        exclude: /(node_modules)|(assets)/,
```
Comment (justinas) [inline web/packages/build/webpack/webpack.base.js:107]: Intentionally removed `assets` as well: according to Ryan, this is probably an unnecessary leftover from way back.

File: web/packages/build/webpack/webpack.base.js [hunk 2]
```diff
@@ -78,7 +78,7 @@ const configFactory = {
       return {
         test: /\.svg$/,
         type: 'asset/inline',
-        exclude: /node_modules/,
+        exclude: /[\\/]node_modules[\\/]/,
```
Comment (hatched) [inline web/packages/build/webpack/webpack.base.js:81]: Is this change necessary? As this will technically introduce a regression from the previous...

| old | new | path | 
|---|---|---|
| ✔️  | ✖️   | node_modules/foo/bar |
| ✔️  | ✔️  | /node_modules/foo/bar |
| ✔️  | ✔️  | foo/bar/node_modules/foo/bar |
| ✔️  | ✔️   | /foo/bar/node_modules/foo/bar |

Comment (justinas) [reply web/packages/build/webpack/webpack.base.js:81]: Webpack seems to match on absolute paths, can be verified via e.g.:

```diff
diff --git a/e b/e
index 04605151e7..db056f5f5f 160000
--- a/e
+++ b/e
@@ -1 +1 @@
-Subproject commit 04605151e7be43fb6831764153cf485e3a6a493f
+Subproject commit db056f5f5fbf96d1fd99c947d8f5be9070715363
diff --git a/web/packages/build/webpack/webpack.base.js b/web/packages/build/webpack/webpack.base.js
index 2806ab488d..ff7b79603d 100644
--- a/web/packages/build/webpack/webpack.base.js
+++ b/web/packages/build/webpack/webpack.base.js
@@ -104,7 +104,10 @@ const configFactory = {
     jsx() {
       return {
         test: /\.(ts|tsx|js|jsx)$/,
-        exclude: /[\\/]node_modules[\\/]/,
+        exclude: (f) => {
+          console.log(f)
+          return f.match(/[\\/]node_modules[\\/]/)
+        },
         use: [
           {
             loader: 'babel-loader',
```

which outputs:

```
/home/justinas/gravitational/teleport/web/packages/teleport/src/boot.tsx
```

This is how #21662 happened in the first place (the dir containing the `assets` substring was above the repo root).

So there can never be a bare `node_modules` at the start (will always be at least `C:\node_modules` or `/node_modules`).

**Review by ryanclark — APPROVED** [review]

**Review by hatched — APPROVED** [review]
